airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py

```diff
@@ -526,50 +927,90 @@ def request_body_json(
+            records = query_job.read_records(sync_mode=SyncMode.full_refresh)
+            for record in records:
+                yield record
+                self._updated_state(self.state, record)
+        else:
+            for row in rows:
+                data = row.get("f")
+                formated_data = {
+                    "_bigquery_table_id": job_id,
+                    "_bigquery_created_time": datetime.now(tz=pytz.timezone("UTC")).isoformat(timespec='microseconds'), #TODO: Update this to row insertion time
+                    **{CHANGE_FIELDS.get(element["name"], element["name"]): SchemaHelpers.format_field(data[fields.index(element)]["v"], element["type"]) for element in fields},
+                }
+                yield formated_data
+                self._updated_state(self.state, formated_data)
 
+    def should_retry(self, response: requests.Response) -> bool:
+        """
+        Override to set different conditions for backoff based on the response from the server.
 
-class TableChangeHistory(BigqueryCDCStream):
-    """  
-    """ 
+        By default, back off on the following HTTP response statuses:
+         - jobComplete false indicates query job not completed
+
+        Unexpected but transient exceptions (connection timeout, DNS resolution failed, etc..) are retried by default.
+        """
+        retry = False
+        records = response.json()
+        if records.get("jobComplete", None):
+            retry = (records["jobComplete"] == False)
+        return retry
+
+
+class TableChangeHistory(BigqueryResultStream):
+    """
+    """
     primary_key = None
-    
-    def __init__(self, project_id: list, dataset_id: str, table_id: str, **kwargs):
+
+    def __init__(self, project_id: list, dataset_id: str, table_id: str, given_name=None, \
+                 where_clause: str="", fallback_start: datetime=None, slice_range=SLICE_RANGE, **kwargs):
         self.project_id = project_id
-        self.parent_stream = dataset_id + "." + table_id
-        super().__init__(self.path(), self.parent_stream, self.get_json_schema, **kwargs)
-        self.stream_obj = BigqueryCDCStream(self.path(), self.parent_stream, self.get_json_schema, **kwargs)
+        self.given_name = given_name
+        self.table_qualifier = dataset_id + "." + table_id
+        self.dataset_id = dataset_id
+        self.table_id = table_id
+        self.where_clause = where_clause.replace("\"", "'")
+        super().__init__(self.path(), self.table_qualifier, self.project_id, self.get_json_schema, retry_policy=self.should_retry, **kwargs)
+        self.stream_obj = BigqueryCDCStream(project_id, dataset_id, table_id, self.path(), self.get_json_schema, \
+                                            given_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, **kwargs)
+
+    @property
+    def namespace(self):
+        return None # TODO: update to self.dataset_id
```

Comment:
What is needed to fix this todo?

Reply:
This is not necessary, as I just wanted to do the same as the old Bigquery connector.